starlette/testclient.py

```diff
@@ -463,9 +463,7 @@ def __exit__(self, *args: typing.Any) -> None:
     async def wait_startup(self) -> None:
         await self.receive_queue.put({"type": "lifespan.startup"})
         message = await self.send_queue.get()
-        if message is None:
-            self.task.result()
-        assert message["type"] == "lifespan.startup.complete"
+        assert message["type"] in ("lifespan.startup.complete", "lifespan.startup.failed")
```

Comment:
In the failure case I *think* we should probably wait for another message (which will be `None` as the sentinel value for "all done on this messaging queue now) and then call `self.task.result()` (which will raise an exception)

Tho we'd want to make sure that we're verifying this with a test case that ensures that startup failures bubble out correctly.

Reply:
Okay, I think I've got it fixed. The tests already verify that the `RuntimeError` raised in the startup callback bubbles out of the app, so I think we're good. Let me know!